File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.java
class Solution {
    public List<Integer> findDisappearedNumbers(int[] nums) {
        List<Integer> ans = new ArrayList<>();
        for(int num: nums){
            nums[Math.abs(num)-1] = -Math.abs(nums[Math.abs(num)-1]); 
        }
        // for(int num:nums){
        //     System.out.println(num);
        // }
        for(int i = 0; i< nums.length; i++){
            if(nums[i] > 0){
                ans.add(i+1);
            }
        }
        return ans;
    }
}

File: 0448-find-all-numbers-disappeared-in-an-array/0448-find-all-numbers-disappeared-in-an-array.py
class Solution:
    def findDisappearedNumbers(self, nums: List[int]) -> List[int]:
        n = len(nums)
        st = set(nums)
        res = []
        for i in range(1,len(nums)+1):
            if i not in st:
                res.append(i)
        return res